apps/backend/src/controllers/agri/bap/onInit.ts

```diff
@@ -0,0 +1,124 @@
+import { NextFunction, Request, Response } from "express";
+import fs from "fs";
+import path from "path";
+import YAML from "yaml";
+import { v4 as uuidv4 } from "uuid";
+import { AGRI_EXAMPLES_PATH, SERVICES_EXAMPLES_PATH, checkIfCustomized, quoteCreatorHealthCareService, redisFetchFromServer, responseBuilder } from "../../../lib/utils";
+
+
+export const onInitController = async (req: Request, res: Response, next: NextFunction) => {
+	try{
+		const on_search = await redisFetchFromServer("on_search", req.body.context.transaction_id);
+		const providersItems = on_search?.message?.catalog["bpp/providers"][0]?.items;
+		req.body.providersItems = providersItems
+		if (checkIfCustomized(req.body.message.order.items)) {
+			return onInitServiceCustomizedController(req, res, next);
+		}
+		onInitConsultationController(req, res, next)
+	}catch(error){
+		return next(error)
+	}
+};
+
+const onInitConsultationController = (req: Request, res: Response, next: NextFunction) => {
+	try{
+		const { context, message: { order: { provider, locations, items, billing, fulfillments, payments,quote } } } = req.body;
+		const { stops, ...remainingfulfillments } = fulfillments[0]
+	
+		const file = fs.readFileSync(
+			path.join(AGRI_EXAMPLES_PATH, "confirm/confirm.yaml")
+		);
+		
+		const response = YAML.parse(file.toString());
+		const timestamp = new Date().toISOString();
+		const responseMessage = {
+			order: {
+				id: uuidv4(),
+				status: "Created",
+				provider: {
+					...provider,
+					...locations
+				},
+				items,
+				billing,
+				fulfillments,
+				quote: quote,
+				payments,
+				created_at: timestamp,
+				updated_at: timestamp,
+			}
+		}
+	
+		return responseBuilder(
+			res,
+			next,
+			context,
+			responseMessage,
+			`${context.bpp_uri}${context.bpp_uri.endsWith("/") ? "confirm" : "/confirm"
+			}`,
+			`confirm`,
+			"agri"
+		);
+	}catch(error){
+		next(error)
+	}
+
+	
+};
+
+const onInitServiceCustomizedController = (req: Request, res: Response, next: NextFunction) => {
+	const { context, providersItems, message: { order: { provider, locations, items, billing, fulfillments, payments } } } = req.body;
```

Comment:
Same here and in all instances. Please also check the confirm controller for this.